testing/test_pytester.py

```diff
@@ -263,14 +263,15 @@ def test_inline_run_sys_modules_snapshot_restore_preserving_modules(
     def test_external_test_module_imports_not_cleaned_up(self, testdir):
         testdir.syspathinsert()
         testdir.makepyfile(imported="data = 'you son of a silly person'")
-        import imported
-
         test_mod = testdir.makepyfile(
             """
             def test_foo():
                 import imported
                 imported.data = 42"""
         )
+
+        import imported
+
```

Comment:
so this is a little bit weird 🤔 

the `import` statement has a side-effect of populating the `importlib` caches, but since the python file adjacent to `imported.py` is created afterwards, calls to look it up will always fail.

by moving the import (or calling `importlib.invalidate_caches()`) this import can succeed again

Reply:
I actually think I can move it back now that I put `importlib.invalidate_caches()` into `pytester` -- let me try that

Reply:
ended up factoring this out, yay.